### PR TITLE
Fix issue 599: Query with different time format does not work if database is not specified.

### PR DESCRIPTION
**Pull request by fredo994**
Fix for issue #599 - to use database information present in the client if not set in query.

I also changed TimeUtils#toTimePrecision to be in compliance with documentation.

### CODE REVIEW

File: src/main/java/org/influxdb/impl/InfluxDBImpl.java [hunk 1]
```diff
@@ -926,6 +921,11 @@ public void dropRetentionPolicy(final String rpName, final String database) {
     executeQuery(this.influxDBService.postQuery(Query.encode(queryBuilder.toString())));
   }
 
+  private String getDatabase(final Query query) {
+  	String db = query.getDatabase();
+  	return db == null ? this.database : db;
```
Comment (majst01) [inline src/main/java/org/influxdb/impl/InfluxDBImpl.java]: I dont like elvis, please consider without

File: src/main/java/org/influxdb/impl/TimeUtil.java [hunk 2]
```diff
@@ -60,7 +60,7 @@ public static String toTimePrecision(final TimeUnit t) {
     case MICROSECONDS:
       return "u";
     case NANOSECONDS:
-      return "n";
+      return "ns";
```
Comment (majst01) [inline src/main/java/org/influxdb/impl/TimeUtil.java]: unrelated change, if this is a problem, please raise another Issue/PR

File: src/main/java/org/influxdb/impl/InfluxDBImpl.java (hunk 1 above)
Comment (fredo994) [reply src/main/java/org/influxdb/impl/InfluxDBImpl.java]: ok

File: src/main/java/org/influxdb/impl/TimeUtil.java (hunk 2 above)
Comment (fredo994) [reply src/main/java/org/influxdb/impl/TimeUtil.java]: This is not a problem (it works in both cases), but I've notice that the code is not consistent with documentation.

**Comment by majst01** [comment]
Checkstyle complains:

```
[ERROR] /usr/src/mymaven/src/main/java/org/influxdb/impl/InfluxDBImpl.java:925:3: File contains tab characters (this is the first instance). [FileTabCharacter]
```

File: src/test/java/org/influxdb/InfluxDBTest.java [hunk 3]
```diff
@@ -39,6 +40,7 @@
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.LongAdder;
 import java.util.function.Consumer;
+import org.junit.runner.notification.RunListener;
```
Comment (majst01) [inline src/test/java/org/influxdb/InfluxDBTest.java]: Unused ?

Comment (fredo994) [reply src/test/java/org/influxdb/InfluxDBTest.java]: Sorry don't know how that got in there.

**Comment by majst01** [comment]
looks good so far, can you please add a entry in the README to point to this ?
